### PR TITLE
Harden runner edge cases

### PR DESCRIPTION
**Pull request by ShiroKSH**
## Summary
- Record setup failures as test failures instead of letting the runner abort or leave test environments mutated.
- Avoid duplicate async results, preserve runner order across files, and reset accumulated results between runs.
- Tighten expectation and loader edge cases, including shared client tests and include-error reporting.
- Fix stale Docker/docs examples that pointed at the wrong artifact layout or obsolete settings.

## Testing
- `git diff --check`
- Checked new GLua self-test files for trailing whitespace, conflict markers, and tabs.

### CODE REVIEW

**Comment by ShiroKSH** [comment]
@sarahsturgeon - could you take a look at the runner edge-case hardening in this PR when you have a chance, or route it to the right reviewer? I can make any requested changes or add tests. Thanks!

**Comment by ShiroKSH** [comment]
@sarahsturgeon Friendly reminder: we'd appreciate it if you could take a look at this pull request when you have a chance.

File: lua/tests/gluatest/runner/test_case_runner.lua [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        state = {}
+    }
+end
+
+return {
+    groupName = "TestCaseRunner",
+
+    cases = {
+        {
+            name = "Records sync beforeEach errors as failures",
```
Comment (A1steaksa) [inline lua/tests/gluatest/runner/test_case_runner.lua]: Case name is difficult to understand unless you already know what the case is and does

File: lua/tests/gluatest/runner/test_case_runner.lua [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+                expect( completed ).to.beTrue()
+                expect( caseRan ).to.beFalse()
+                expect( #testGroupRunner.results ).to.equal( 1 )
+                expect( testGroupRunner.results[1].success ).to.beFalse()
+                expect( testGroupRunner.results[1].errInfo.reason ).to.equal( "Unhandled: sync beforeEach failed" )
+            end
+        },
+        {
+            name = "Records async beforeEach errors as failures",
```
Comment (A1steaksa) [inline lua/tests/gluatest/runner/test_case_runner.lua]: Case name is difficult to understand unless you already know what the case is and does

File: lua/tests/gluatest/loader/checkSendToClients.lua [hunk 3]
```diff
@@ -65,6 +65,25 @@ return {
...
+            cleanup = function( state )
+                GLuaTest.RunClientsideConVar:SetBool( state.currentRunClientside )
+            end
+        },
+
+        {
+            name = "Sends shared cases when the gluatest_client_enable ConVar is enabled",
+            func = function( state )
+                local Loader = state.Loader
+                state.currentRunClientside = GLuaTest.RunClientsideConVar:GetBool()
```
Comment (A1steaksa) [inline lua/tests/gluatest/loader/checkSendToClients.lua]: Nit: This variable stores the _prior_ `RunClientsideConVar` value, not the _current_ value

File: lua/gluatest/expectations/negative.lua [hunk 4]
```diff
@@ -178,7 +178,7 @@ return function( subject, ... )
     function expectations.succeed()
         assert( TypeID( subject ) == TYPE_FUNCTION, ".succeed expects a function" )
 
-        local success = pcall( subject, unpack( args ) )
+        local success = pcall( subject, unpack( args, 1, args.n ) )
```
Comment (A1steaksa) [inline lua/gluatest/expectations/negative.lua]: These additions to `unpack` throughout the PR strike me as unnecessary, given that the defaults for `unpack` should produce these same results

File: lua/gluatest/expectations/negative.lua [hunk 5]
```diff
@@ -202,10 +202,10 @@ return function( subject, ... )
         assert( TypeID( subject ) == TYPE_FUNCTION, ".errWith expects a function" )
         assert( isstring( comparison ), "errWith expects a string" )
 
-        local success, err = pcall( subject, unpack( args ) )
+        local success, err = pcall( subject, unpack( args, 1, args.n ) )
 
         if success == true then
-            i.expected( "to error" )
+            return
```
Comment (A1steaksa) [inline lua/gluatest/expectations/negative.lua]: This `if` condition should be inverted to remove the case that just calls `return` to remove the "empty" code path

**Review by A1steaksa — CHANGES_REQUESTED** [review]
These seem like good changes.  There are a few small things I've noted, but generally I think this is an improvement.

Thanks for taking the time to contribute to GLuaTest!

File: lua/tests/gluatest/runner/test_case_runner.lua (hunk 1 above)
Comment (ShiroKSH) [reply lua/tests/gluatest/runner/test_case_runner.lua]: Renamed this case to make the expected behavior explicit: Fails the case when beforeEach errors. See lua/tests/gluatest/runner/test_case_runner.lua in efd921b.

File: lua/tests/gluatest/runner/test_case_runner.lua (hunk 2 above)
Comment (ShiroKSH) [reply lua/tests/gluatest/runner/test_case_runner.lua]: Renamed this case to make the async behavior explicit: Fails the async case when beforeEach errors. See lua/tests/gluatest/runner/test_case_runner.lua in efd921b.

File: lua/tests/gluatest/loader/checkSendToClients.lua (hunk 3 above)
Comment (ShiroKSH) [reply lua/tests/gluatest/loader/checkSendToClients.lua]: Renamed state.currentRunClientside to state.previousRunClientside throughout the loader self-test file in efd921b.

File: lua/gluatest/expectations/negative.lua (hunk 4 above)
Comment (ShiroKSH) [reply lua/gluatest/expectations/negative.lua]: Removed the unpack arity change and dropped the trailing-nil self-tests in efd921b.

File: lua/gluatest/expectations/negative.lua (hunk 5 above)
Comment (ShiroKSH) [reply lua/gluatest/expectations/negative.lua]: Inverted the branch so errWith only handles the failing call path; successful calls now fall through in efd921b.

**Comment by ShiroKSH** [comment]
Hi maintainers, this PR has been open for over two weeks. When you have a moment, I would appreciate a review. Thank you.

**Comment by ShiroKSH** [comment]
@A1steaksa All five inline requests were applied in efd921b5: the two case names are explicit, the saved ConVar state uses `previousRunClientside`, the unnecessary `unpack` arity changes and tests are removed, and the `errWith` branch is inverted. All five conversations are resolved. When convenient, could you re-review the current head so the stale `changes requested` state can be cleared?
